File: 28-06-2025_Counting_elements_in_two_arrays.py
class Solution:
    def countLessEq(self, a, b):
        # code here
    
        max_val = max(max(a), max(b))  # Step 1
    
        freq = [0] * (max_val + 2)     # Step 2
        for num in b:
            freq[num] += 1

    # Step 3: Build prefix sum
        for i in range(1, len(freq)):
            freq[i] += freq[i - 1]

    # Step 4: Query prefix sum
        result = [freq[x] for x in a]
        return result

File: 28-06-2025_Counting_elements_in_two_arrays.cpp
class Solution {
  public:
    vector<int> countLessEq(vector<int>& a, vector<int>& b) {
        // code here
        int maxVal = 0;
    for (int num : a) maxVal = max(maxVal, num);
    for (int num : b) maxVal = max(maxVal, num);

    
    vector<int> freq(maxVal + 2, 0);
    for (int num : b) {
        freq[num]++;
    }

   
    for (int i = 1; i <= maxVal + 1; ++i) {
        freq[i] += freq[i - 1];
    }

   
    vector<int> result;
    for (int x : a) {
        result.push_back(freq[x]);
    }

    return result;
    }
};
